src/pattern.rs

```diff
@@ -10,10 +10,10 @@ pub struct RegexSearcher<'r, 't> {
     next_match: Option<(usize, usize)>,
 }
 
-impl<'r, 't> Pattern<'t> for &'r Regex {
-    type Searcher = RegexSearcher<'r, 't>;
+impl<'r> Pattern for &'r Regex {
+    type Searcher<'t> = RegexSearcher<'r, 't>;
 
-    fn into_searcher(self, haystack: &'t str) -> RegexSearcher<'r, 't> {
+    fn into_searcher<'h>(self, haystack: &'h str) -> RegexSearcher<'r, 'h> {
```

Comment:
Sorry, can this be `'t` instead of `'h` to be consistent? I don't think there is a shadowing concern here.